Contest_API.md

```diff
@@ -338,29 +338,30 @@ for these, since there will always be another event sent.
 
 The events are served as JSON objects, with every event corresponding to
 a change in a single object (submission, judgement, language, team,
-etc.) or full endpoint. The general format for events is:
+etc.) or entire collection. The general format for events is:
 
 ```json
-{"contest_id": "<id>", "endpoint": "<endpoint>", "id": "<id>", "data": <JSON data for element> }
+{"type": "<type>", "id": "<id>", "data": <JSON data for element> }
 ```
 
-| Name        | Type   | Required? | Nullable? | Description
-| :---------- | :----- | :-------- | :-------- | :----------
-| contest\_id | string | yes       | no        | The contest id.
-| endpoint    | string | yes       | yes       | The API endpoint, i.e. type of contest object above. Can be used for filtering.
-| id          | string | yes       | yes       | The id of the object that changed.
-| data        | object | yes       | yes       | The data is the object that would be returned if calling the corresponding API endpoint at this time, i.e. an object or null for deletions.
+| Name        | Type            | Required? | Nullable? | Description
+| :---------- | :-------------- | :-------- | :-------- | :----------
+| type        | string          | yes       | no        | The type of contest object that changed. Can be used for filtering.
+| id          | string          | yes       | yes       | The id of the object that changed, or null for the entire collection/singleton.
+| data        | array or object | yes       | yes       | The updated value, i.e. what would be returned if calling the corresponding API endpoint at this time: an array, object, or null for deletions.
 
+Each event is a notification that an object or a collection has changed
+(and hence the contents of the corresponding endpoint) to `data`.
 
-The meaning of an event is to say that the contents at endpoint
-`/contests/<contest_id>/<endpoint>/<id>` now has the contents of `data`.
+If `type` is `contest`, then the contest at `/contests/<id>` now has the contents of `data`.
+If `id` is not null, then the object at `/contests/<contest_id>/<type>/<id>` now has the contents of `data`.
+If `id` is null, then the entire collection at `/contests/<contest_id>/<type>` now has the contents of `data`.
```

Comment:
Allowed values for `type` is not specified anywhere. The above three rows imply that `contest` or any of the endpoints under `/contests/<contest_id>/` are allowed, which is almost true.

I'm not sure if this must be fixed?